packages/wrangler/src/api/pages/create-worker-bundle-contents.ts

```diff
@@ -12,28 +13,25 @@ import type { FormData } from "undici";
  * contents
  */
 export async function createUploadWorkerBundleContents(
-	workerBundle: BundleResult
+	workerBundle: BundleResult,
+	config: Config | undefined
 ): Promise<Blob> {
-	const workerBundleFormData = createWorkerBundleFormData(workerBundle);
+	console.log(config);
+	const workerBundleFormData = createWorkerBundleFormData(workerBundle, config);
 	const metadata = JSON.parse(workerBundleFormData.get("metadata") as string);
 
-	/**
-	 * Pages doesn't need the metadata bindings returned by
-	 * `createWorkerBundleFormData`. Let's strip them out and return only
-	 * the contents we need
-	 */
-	workerBundleFormData.set(
-		"metadata",
-		JSON.stringify({ main_module: metadata.main_module })
-	);
+	workerBundleFormData.set("metadata", JSON.stringify(metadata));
```

Comment:
I think these lines are no longer needed. They were only here because we were trying to strip out the bindings from the metadata.

Reply:
In fact perhaps this whole function is redundant now?

Reply:
I've modified this function to strip out the empty bindings key from the metadata if no config has been found